### PR TITLE
[HOLD 2.7.0] Add pandas to nightly wheel installs

### LINKED ISSUE #4238: [future] NINF removal in python 2.0 causing cron ci to fail
Cron CI is failing because of namspace changes affecting the use numpy.NINF in `pandas` (yes it's an accidental dependency it seems, not sure exactly from where yet).

Fix has been made upstream, we just need to add `pandas` to our list of future installs.

### PR DESCRIPTION
**Pull request by IAlibay**
Fixes #4238 

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
:books: Documentation preview :books:: https://mdanalysis--4239.org.readthedocs.build/en/4239/

### CODE REVIEW

File: .github/workflows/gh-ci-cron.yaml
Comment (IAlibay) [inline .github/workflows/gh-ci-cron.yaml]: Remove before merge

**Comment by IAlibay** [comment]
Logging issue happens whether or not nightly pandas is installed, looks like that's a separate isssue.

**Comment by IAlibay** [comment]
Timeouts aside, this should be good to go if anyone wants to review - CI is returning green for `numpy_and_scipy_dev`  (https://github.com/MDAnalysis/mdanalysis/actions/runs/5861485404/job/15891653149?pr=4239) so I'm truning off the PR trigger.

**Review by fiona-naughton — APPROVED** [review]
lgtm, thanks @IAlibay 

(If I'm following correctly, is this something where we expect to be able come back remove pandas later, after figuring out the accidental-dependency?)

**Comment by IAlibay** [comment]
> (If I'm following correctly, is this something where we expect to be able come back remove pandas later, after figuring out the accidental-dependency?)

Yeah eventually, it's probably being pulled in by one of the core dependencies we've been wanting to get rid of (biopython, networkx, or matplotlib).

P.S. Thanks for the review, I won't merge until we get 2.7.0 started.

**Pull request merged**
